sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -55,7 +56,7 @@ object SQLConf {
   val WAREHOUSE_PATH = SQLConfigBuilder("spark.sql.warehouse.dir")
     .doc("The default location for managed databases and tables.")
     .stringConf
-    .createWithDefault("${system:user.dir}/spark-warehouse")
+    .createWithDefault("${system:user.dir}" + File.separator + "spark-warehouse")
```

Comment:
Yeah, we can do it in that way. I am just afraid it might be not always right. Below are two related discussions: https://en.wikipedia.org/wiki/File_URI_scheme#Meaning_of_slash_character
http://stackoverflow.com/questions/1131273/java-file-touri-tourl-on-windows-file?answertab=votes#tab-top

Let me know which way is preferred.